File: app/src/main/java/de/openfiresource/falarm/models/database/OperationMessageDao.java
package de.openfiresource.falarm.models.database;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import java.util.List;

import io.reactivex.Flowable;

@Dao
public interface OperationMessageDao {

    @Query("SELECT * FROM operation_message ORDER BY timestamp DESC")
    LiveData<List<OperationMessage>> getAllAsync();

    @Query("SELECT * FROM operation_message ORDER BY timestamp DESC")
    List<OperationMessage> getAll();

    @Query("SELECT * FROM operation_message WHERE id = :id LIMIT 1")
    OperationMessage findById(long id);

    @Query("SELECT * FROM operation_message WHERE id = :id LIMIT 1")
    Flowable<OperationMessage> findByIdAsync(long id);

    @Insert
    long insertOperationMessage(OperationMessage operationMessage);

    @Update
    void updateOperationMessage(OperationMessage operationMessage);

    @Delete
    void deleteOperationMessage(OperationMessage operationMessage);
}

File: app/src/main/java/de/openfiresource/openpager/models/database/OperationMessageDao.kt
package de.openfiresource.openpager.models.database

import android.arch.lifecycle.LiveData
import android.arch.persistence.room.*
import io.reactivex.Flowable

@Dao
interface OperationMessageDao {

    @Query("SELECT * FROM operation_message ORDER BY timestamp DESC")
    fun allAsync(): LiveData<List<OperationMessage>>

    @Query("SELECT * FROM operation_message ORDER BY timestamp DESC")
    fun all(): List<OperationMessage>

    @Query("SELECT * FROM operation_message WHERE id = :id LIMIT 1")
    fun findById(id: Long): OperationMessage

    @Query("SELECT * FROM operation_message WHERE id = :id LIMIT 1")
    fun findByIdAsync(id: Long): Flowable<OperationMessage>

    @Insert
    fun insertOperationMessage(operationMessage: OperationMessage): Long

    @Update
    fun updateOperationMessage(operationMessage: OperationMessage)

    @Delete
    fun deleteOperationMessage(operationMessage: OperationMessage)
}
